object.go

```diff
@@ -14,11 +14,18 @@
 
 package openssl
 
-// #include "shim.h"
+// #include <openssl/objects.h>
 import "C"
+import "unsafe"
 
 // CreateObjectIdentifier creates ObjectIdentifier and returns NID for the created
 // ObjectIdentifier
 func CreateObjectIdentifier(oid string, shortName string, longName string) NID {
-	return NID(C.OBJ_create(C.CString(oid), C.CString(shortName), C.CString(longName)))
+	cOid := C.CString(oid)
```

Comment:
I'm guessing we don't run any of the tests in this repo under valgrind currently? If not, probably worth setting that up -- writing/using C bindings without any memory checking is a dangerous game...

Reply:
No we don't, I manually went through to audit all the places we're calling CString to ensure we free them at a later date.

Move to a new issue?

Reply:
https://github.com/pexip/go-openssl/issues/6